# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/shared/CodeGenerator-shared.h
# Commit: 659b24c3626c
# Full Hash: 659b24c3626c6a299ef34ce4a0109fca2e2a9112
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 21:38:12
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.4: Add tls dependency for WasmTruncateToInt64 and Int64ToFloatingPoint for arm. r=lth
#   
#   In this patch we add a tls dependency for the remaining nodes which use
#   BuiltinThunk to call c++ runtime. By ABI requirements WasmTlsReg should
#   be set.
# ==============================================================================

diff -r 473f1cb675ea -r 659b24c3626c js/src/jit/shared/CodeGenerator-shared.h
--- a/js/src/jit/shared/CodeGenerator-shared.h	Fri Sep 11 05:37:42 2020 +0000
+++ b/js/src/jit/shared/CodeGenerator-shared.h	Fri Sep 11 12:02:42 2020 +0000
@@ -551,6 +551,16 @@
         flags_(mir->flags()),
         bytecodeOffset_(mir->bytecodeOffset()) {}
 
+  OutOfLineWasmTruncateCheckBase(MWasmBuiltinTruncateToInt64* mir,
+                                 FloatRegister input, Register64 output)
+      : fromType_(mir->input()->type()),
+        toType_(MIRType::Int64),
+        input_(input),
+        output_(Register::Invalid()),
+        output64_(output),
+        flags_(mir->flags()),
+        bytecodeOffset_(mir->bytecodeOffset()) {}
+
   OutOfLineWasmTruncateCheckBase(MWasmTruncateToInt64* mir, FloatRegister input,
                                  Register64 output)
       : fromType_(mir->input()->type()),